aten/src/ATen/native/transformers/cuda/sdp_utils.h

```diff
@@ -232,14 +259,15 @@ inline bool use_mem_efficient_attention(sdp_params params, bool debug) {
       at::kHalf, at::kFloat, at::kBFloat16};
 
   //  Define gate functions that determine if a flash kernel can be ran
-  std::vector<std::function<bool(sdp_params, bool)>> constraints{
+  constexpr std::array<bool(*)(sdp_params, bool), 8> constraints{{
       check_gpu_sm50_or_greater,
       check_runtime_disabled_mem_efficient,
+      check_requires_grad_and_nested,
```

Comment:
Since the caller is composite we add a check to ensure that if the input is nested and requires grad we don't return fused kernel.